applications/MeshMovingApplication/tests/cpp_tests/test_impose_mesh_motion_process.cpp

```diff
@@ -0,0 +1,163 @@
+        GeometryType(domain_corners),
+        rModelPart,
+        mesh_generator_parameters
+    ).Execute();
+
+    // Add DoFs
+    for (auto& r_node : rModelPart.Nodes()) {
+        r_node.AddDof(DISPLACEMENT_X);
+        r_node.AddDof(DISPLACEMENT_Y);
+        r_node.AddDof(DISPLACEMENT_Z);
+    }
+}
+
+
+void CheckNodes(const ModelPart& rModelPart)
+{
+    Node<3> node;
+    const double tolerance = 1e-10;
+
+    KRATOS_CHECK(rModelPart.Nodes().size() == 9);
+
+    node = rModelPart.GetNode(1);
+    KRATOS_CHECK_NEAR(node.GetValue(MESH_DISPLACEMENT_X), 0.00, tolerance);
+    KRATOS_CHECK_NEAR(node.GetValue(MESH_DISPLACEMENT_Y), 2.00, tolerance);
+    KRATOS_CHECK_NEAR(node.GetValue(MESH_DISPLACEMENT_Z), 4.00, tolerance);
+
+    node = rModelPart.GetNode(2);
+    KRATOS_CHECK_NEAR(node.GetValue(MESH_DISPLACEMENT_X), 0.00, tolerance);
+    KRATOS_CHECK_NEAR(node.GetValue(MESH_DISPLACEMENT_Y), 2.00, tolerance);
+    KRATOS_CHECK_NEAR(node.GetValue(MESH_DISPLACEMENT_Z), 4.00, tolerance);
+
+    node = rModelPart.GetNode(3);
+    KRATOS_CHECK_NEAR(node.GetValue(MESH_DISPLACEMENT_X), 0.00, tolerance);
+    KRATOS_CHECK_NEAR(node.GetValue(MESH_DISPLACEMENT_Y), 2.00, tolerance);
+    KRATOS_CHECK_NEAR(node.GetValue(MESH_DISPLACEMENT_Z), 4.00, tolerance);
+
+    node = rModelPart.GetNode(4);
+    KRATOS_CHECK_NEAR(node.GetValue(MESH_DISPLACEMENT_X), -1.0, tolerance);
+    KRATOS_CHECK_NEAR(node.GetValue(MESH_DISPLACEMENT_Y), 2.00, tolerance);
+    KRATOS_CHECK_NEAR(node.GetValue(MESH_DISPLACEMENT_Z), 5.00, tolerance);
+
+    node = rModelPart.GetNode(5);
+    KRATOS_CHECK_NEAR(node.GetValue(MESH_DISPLACEMENT_X), -1.0, tolerance);
+    KRATOS_CHECK_NEAR(node.GetValue(MESH_DISPLACEMENT_Y), 2.00, tolerance);
+    KRATOS_CHECK_NEAR(node.GetValue(MESH_DISPLACEMENT_Z), 5.00, tolerance);
+
+    node = rModelPart.GetNode(6);
+    KRATOS_CHECK_NEAR(node.GetValue(MESH_DISPLACEMENT_X), -1.0, tolerance);
+    KRATOS_CHECK_NEAR(node.GetValue(MESH_DISPLACEMENT_Y), 2.00, tolerance);
+    KRATOS_CHECK_NEAR(node.GetValue(MESH_DISPLACEMENT_Z), 5.00, tolerance);
+
+    node = rModelPart.GetNode(7);
+    KRATOS_CHECK_NEAR(node.GetValue(MESH_DISPLACEMENT_X), -2.0, tolerance);
+    KRATOS_CHECK_NEAR(node.GetValue(MESH_DISPLACEMENT_Y), 2.00, tolerance);
+    KRATOS_CHECK_NEAR(node.GetValue(MESH_DISPLACEMENT_Z), 6.00, tolerance);
+
+    node = rModelPart.GetNode(8);
+    KRATOS_CHECK_NEAR(node.GetValue(MESH_DISPLACEMENT_X), -2.0, tolerance);
+    KRATOS_CHECK_NEAR(node.GetValue(MESH_DISPLACEMENT_Y), 2.00, tolerance);
+    KRATOS_CHECK_NEAR(node.GetValue(MESH_DISPLACEMENT_Z), 6.00, tolerance);
+
+    node = rModelPart.GetNode(9);
+    KRATOS_CHECK_NEAR(node.GetValue(MESH_DISPLACEMENT_X), -2.0, tolerance);
+    KRATOS_CHECK_NEAR(node.GetValue(MESH_DISPLACEMENT_Y), 2.00, tolerance);
+    KRATOS_CHECK_NEAR(node.GetValue(MESH_DISPLACEMENT_Z), 6.00, tolerance);
+}
+
+
+KRATOS_TEST_CASE_IN_SUITE(ImposeMeshMotionProcessRotationAxis, MeshMovingApplicationFastSuite)
```

Comment:
as mentioned in your other PR, I would remove the c++ test if it does the same as the python one
My own rule of thumb is to use C++ for unit tests / things that cannot be tested in python and python for integration tests/larger tests that test entire workflows

My motivation is not to unnecessarily increase compile times